lighthouse-core/report/v2/renderer/details-renderer.js

```diff
@@ -134,3 +197,21 @@ DetailsRenderer.ListDetailsJSON; // eslint-disable-line no-unused-expressions
  * }}
  */
 DetailsRenderer.CardsDetailsJSON; // eslint-disable-line no-unused-expressions
+
+/** @typedef {{
+ *     type: string,
+ *     header: ({text: string}|undefined),
+ *     items: !Array<!Array<!DetailsRenderer.DetailsJSON>>,
+ *     itemHeaders: !Array<!DetailsRenderer.DetailsJSON>
```

Comment:
`detailsjson` is the base (structural) type that is the superset of all of them. It only has fields all of them have: `type` and optional `text` (really text is there from when they all had `text` and should probably be removed). `render()` takes that and casts according to its `type` to the appropriate type, and we'll take responsibility to make sure that if `type` is set to x, then that object really is of `type` x. It's a hack, but simple (we only have to have humans in the loop checking the top level `details` object), and can easily be reworked once we know the full outlines of the types we need.

Allowing arbitrary nesting means that it's much less simple since we have to guarantee that all the nested items and itemHeaders are really whatever their `type` says. It also means we're implicitly breaking the no-typedef-cycles, just getting around it by casting, which is what I meant by recursive madness.

If you're saying we should have a different type explicitly for leaf nodes to make this better, I think that's great